src/@utils/order.ts

```diff
@@ -48,57 +47,153 @@ export async function order(
     }
   } as OrderParams
 
-  // TODO: we need to approve provider fee
+  return {
+    initializeData,
+    orderParams
+  }
+}
+
+/**
+ * Get and execute transaction based on asset type
+ * @param web3
+ * @param asset
+ * @param accountId
+ * @param action
+ * @param orderPriceAndFees
+ * @returns {TransactionReceipt} receipt of the order / gas estimate
+ */
+async function getTransaction(
```

Comment:
good direction centralizing this, just a note that we need to test all possible download options with this change as PR touches all of them.

Just logically, the estimations are no transactions so not sure this is the best place to hold them too